holisticai/explainability/metrics/feature_importance/extractors/lime_feature_importance.py

```diff
@@ -0,0 +1,165 @@
+import numpy as np
+import pandas as pd
+from lime import lime_tabular
+
+from holisticai.utils._validation import (
+    _array_like_to_series,
+    _matrix_like_to_dataframe,
+)
+
+from ..local_importance._local_metrics import (
+    dataset_spread_stability,
+    features_spread_stability,
+)
+from .extractor_utils import (
+    BaseFeatureImportance,
+    LocalFeatureImportance,
+    get_index_groups,
+    get_top_k_lime,
+)
+
+
+def compute_lime_feature_importance(model_type, model, x, y):
+    if not isinstance(x, pd.DataFrame):
+        x = _matrix_like_to_dataframe(x)
+
+    if not isinstance(y, pd.Series):
+        y = _array_like_to_series(y)
+
+    if model_type == "binary_classification":
+        lime_mode = "classification"
+        scorer = model.predict_proba
+
+    elif model_type == "regression":
+        lime_mode = "regression"
+        scorer = model.predict
+
+    index_groups = get_index_groups(model_type, y)
+    features_importance = lime_creator(
+        scorer=scorer, X=x, index_groups=index_groups, mode=lime_mode
+    )
+    conditional_features_importance = {
+        str(c): gdf for c, gdf in features_importance.groupby("Sample Group")
+    }
+
+    return LimeFeatureImportance(features_importance, conditional_features_importance)
+
+
+def lime_creator(
+    scorer,
+    X,
+    index_groups=None,
+    num_features=None,
+    num_samples=None,
+    mode="classification",
+):
+    """
+    Parameters
+    ----------
+    scorer: sklearn-like scorer
+        scorer function
+    X: np.array
+        input data
+    index_groups: dict
+        dictionary with groups
+    num_features: int
+        number of features to select
+    num_samples: int
+        number of samples to select
+    mode: str
+        classification or regression
+    """
+    # load and do assignment
+    if num_features is None:
+        num_features = np.min([X.shape[1], 50])
+
+    if num_samples is None:
+        num_samples = np.min([X.shape[0], 50])
+
```

Comment:
Would it be worse make the max_num_features and max_num_samples some sort of settings? Also if num_samples and num_features are passed of as arguments (if they are not None), should it still not select the min of the shape of X and that number?